common/signals.go

```diff
@@ -0,0 +1,43 @@
+package common
+
+import (
+	"os"
+	"os/signal"
+	"runtime"
+	"syscall"
+)
+
+// A subsystem/server/... that can be stopped or queried about the status with a signal
+type SignalsReceiver interface {
+	Status() string
+	Stop() error
+}
+
+func InitDefaultSignalsHandler(ss []SignalsReceiver) {
```

Comment:
perhaps make that a varargs param. Also, there's nothing default about this. I suggest `StartSignalHandlers`.